skl2onnx/operator_converters/nearest_neighbours.py

```diff
@@ -5,498 +5,236 @@
 # --------------------------------------------------------------------------
```

Comment:
Why do we need to change the entire KNN converter implementation to support batch prediction?

Reply:
I wanted to reuse the cdist optimisation I made for the GaussianProcess. It was faster that way.